trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmAccessionImporter.php

```diff
@@ -0,0 +1,885 @@
+            // Insert this property into the stockprop table and increment the max rank by one
+            $values = [
+              'stock_id' => $stock_id,
+              'type_id' => $cvterm_id,
+              'value' => $prop_value,
+              'rank' => ++$maxrank
+            ];
+            $result = $this->connection->insert('1:stockprop')
+              ->fields($values)
+              ->execute();
+
+            // If the primary key is not available, then the insert failed
+            if (!$result) {
+              $this->logger->error("Insertion of stock property \"@property\" into chado.stockprop failed.", ['@property' => $property]);
+              $this->error_tracker = TRUE;
+              return false;
+            }
+          }
+        }
+
+        // If no records exist, then insert the property as normal
+        else {
+          $values = [
+            'stock_id' => $stock_id,
+            'type_id' => $cvterm_id,
+            'value' => $prop_value
+          ];
+          $result = $this->connection->insert('1:stockprop')
+            ->fields($values)
+            ->execute();
+
+          // If the primary key is not available, then the insert failed
+          if (!$result) {
+            $this->logger->error("Insertion of stock property \"@property\" into chado.stockprop failed.", ['@property' => $property]);
+            $this->error_tracker = TRUE;
+            return false;
+          }
+        }
+      }
+      else {
+        $this->logger->error("Unable to retrieve the cvterm_id of property \"@property\"", ['@property' => $property]);
+        $this->error_tracker = TRUE;
+        return false;
+      }
+    }
+  }
+
+  /**
+   * Loads each synonym into the chado.synonym and chado.stock_synonym
+   * tables. Returns true if the insert was successful.
+   *
+   * @param int $stock_id
+   *   The value of the primary key for the stock record in Chado.
+   * @param string $stock_properties
+   *   The name that is a synonym of the current germplasm. Multiple
+   *   synonyms may be specified, in which case they are expected to
+   *   be separated using a comma or semicolon.
+   * @return boolean
+   *   Returns true if inserting all the properties was successful,
+   *   including if there are no properties
+   */
+  public function loadSynonyms($stock_id, $synonyms, $organism_id) {
+
+    if ($synonyms) {
+      // Separate out multiple synonyms if we have them by either
+      // semicolons or commas. Whitespace is optional
+      $all_synonyms = preg_split("/[;,]\s*/", $synonyms);
+
+      foreach ($all_synonyms as $synonym) {
+        $synonym = trim($synonym);
+        $synonym_type_id = $this->getCVterm('synonym');
+
+        // Check for and load any synonyms to chado.synonym
+        $synonym_query = $this->connection->select('1:synonym', 's')
+          ->fields('s', ['synonym_id'])
+          ->condition('s.name', $synonym, '=')
+          ->condition('s.type_id', $synonym_type_id, '=');
```

Comment:
Open question: do we want to restrict by type here? Might it make sense to look across synonym types for matches?

Reply:
Well..... I realized that I assumed the type was the cv term for "synonym", but that seems redundant since this is the synonym table after all 😅 The type_id column is required, so I think we should open an issue to discuss further what we want stored here.

Reply:
So this has accidentally started an interesting discussion. We are currently using "synonym" as the type of "synonym" 🙈 This column is required. Others in the community are essentially doing the same thing (i.e. "Alias" is the type for all synonyms according to the Mainlab Chado Loader) to fill in a required column that they don't really need or have any guidance on using 🤷 So, we're going to just leave this as is 🙃 Pretend I said nothing 👼